packages/next/server/base-server.ts

```diff
@@ -985,7 +985,9 @@ export default abstract class Server {
       ...partialContext,
       renderOpts: {
         ...this.renderOpts,
-        supportsDynamicHTML: userAgent ? !isBot(userAgent) : false,
+        supportsDynamicHTML: userAgent
+          ? !isBot(userAgent)
+          : !!this.renderOpts.supportsDynamicHTML,
```

Comment:
It seems strange to fall back to `this.renderOpts.supportsDynamicHTML` here IMO. Safest thing to do is to just use `supportsDynamicHTML: false` which is essentially identical to react@<=17 behavior

Reply:
Leave the `this.renderOpts.supportsDynamicHTML` as fallback because here the logic for crawlers is shared, for web-server it's `true` and for node-server will be `false`. And later it will turn it off it doesn't match the condition